src/Databases/DataLake/ICatalog.h

```diff
@@ -2,43 +2,67 @@
 #include <Core/Types.h>
 #include <Core/NamesAndTypes.h>
 #include <Core/SettingsEnums.h>
-#include <Databases/Iceberg/StorageCredentials.h>
-#include <Databases/Iceberg/DatabaseIcebergStorageType.h>
+#include <Databases/DataLake/StorageCredentials.h>
+#include <Storages/ObjectStorage/StorageObjectStorageSettings.h>
+#include <Databases/DataLake/DatabaseDataLakeStorageType.h>
 
-namespace Iceberg
+namespace DataLake
 {
-using StorageType = DB::DatabaseIcebergStorageType;
+
+using StorageType = DB::DatabaseDataLakeStorageType;
 StorageType parseStorageTypeFromLocation(const std::string & location);
 
+struct DataLakeSpecificMetadata
```

Comment:
```suggestion
struct DataLakeSpecificTableProperties
```
(Too many metadatas, below `TableMetadata`, in another place `DeltaLake(etc)Metadata`, so it becomes confusing)
Also maybe move it inside `TableMetadata` class below? Because it is part of it